Tests/ApolloTests/Network/URLSessionClientTests.swift

```diff
@@ -17,14 +21,38 @@ class URLSessionClientLiveTests: XCTestCase {
     super.tearDown()
```

Comment:
We need to add `sessionConfiguration = nil` here.

The way XCTest works is that an instance of the `XCTestCase` subclass is instantiated for each individual test. These instances are not deallocated after their test finishes. Because they contain all of the necessary information for constructing the test report. So every single instance variable will be retained in memory _multiplied by the number of tests_ until the entire test suite finishes. This is why we `nil` things out in `tearDown`.

Usually this doesn't matter much. The amount of objects you would need to have in order to actually cause memory issues is absurdly high. But it's still best practice. And it does end up mattering when you have things in async tests that can be doing things on background threads and keeping those threads alive past the test's completion.